Sources/NetworkRequestHelper.swift

```diff
@@ -195,79 +193,74 @@ class NetworkRequestHelper {
     /// - SeeAlso:
     ///     - ``setExpectationForNetworkRequest(url:httpMethod:)``
     func assertAllNetworkRequestExpectations(ignoreUnexpectedRequests: Bool = true, file: StaticString = #file, line: UInt = #line) {
-        queue.sync {
-            if expectedNetworkRequests.isEmpty {
-                if !ignoreUnexpectedRequests {
-                    assertUnexpectedRequests(file: file, line: line)
-                }
-                return
+        if expectedNetworkRequests.isEmpty {
+            if !ignoreUnexpectedRequests {
+                assertUnexpectedRequests(file: file, line: line)
             }
+            return
+        }
 
-            for expectedRequest in expectedNetworkRequests {
-                let waitResult = expectedRequest.value.await(timeout: 10)
-                let expectedCount: Int32 = expectedRequest.value.getInitialCount()
-                let receivedCount: Int32 = expectedRequest.value.getInitialCount() - expectedRequest.value.getCurrentCount()
+        for expectedRequest in expectedNetworkRequests {
```

Comment:
@timkimadobe,Since expectedRequest is being modified by design, right now we need to remove the queue sync but in future we might need to modify this logic keeping synchronization in mind.